lib/buffer.js

```diff
@@ -1259,12 +1261,30 @@ function atob(input) {
   if (arguments.length === 0) {
     throw new ERR_MISSING_ARGS('input');
   }
+
   input = `${input}`;
+  let nonAsciiWhitespaceCharCount = 0;
+
   for (let n = 0; n < input.length; n++) {
-    if (!ArrayPrototypeIncludes(kForgivingBase64AllowedChars,
-                                StringPrototypeCharCodeAt(input, n)))
+    const char = StringPrototypeCharCodeAt(input, n);
+
+    if (ArrayPrototypeIncludes(kForgivingBase64AllowedChars, char)) {
+      nonAsciiWhitespaceCharCount++;
+    } else if (!ArrayPrototypeIncludes(asciiWhitespaceCharacters, char)) {
       throw lazyDOMException('Invalid character', 'InvalidCharacterError');
+    }
   }
+
+  // If data's code point length divides by 4 leaving a remainder of 1, then
+  // return failure.
+  //
+  // See #3 - https://infra.spec.whatwg.org/#forgiving-base64
+  if (nonAsciiWhitespaceCharCount % 4 === 1) {
+    throw lazyDOMException(
+      'The string to be decoded is not correctly encoded.',
+      'ValidationError');
```

Comment:
Still interested in feedback on whether I should update this error code to some other value. Thoughts @aduh95 @zloirock?

Reply:
It looks like `DOMException` has the following error codes:

```
INDEX_SIZE_ERR: 1,
DOMSTRING_SIZE_ERR: 2,
HIERARCHY_REQUEST_ERR: 3,
WRONG_DOCUMENT_ERR: 4,
INVALID_CHARACTER_ERR: 5,
NO_DATA_ALLOWED_ERR: 6,
NO_MODIFICATION_ALLOWED_ERR: 7,
NOT_FOUND_ERR: 8,
NOT_SUPPORTED_ERR: 9,
INUSE_ATTRIBUTE_ERR: 10,
INVALID_STATE_ERR: 11,
SYNTAX_ERR: 12,
INVALID_MODIFICATION_ERR: 13,
NAMESPACE_ERR: 14,
INVALID_ACCESS_ERR: 15,
VALIDATION_ERR: 16,
TYPE_MISMATCH_ERR: 17,
SECURITY_ERR: 18,
NETWORK_ERR: 19,
ABORT_ERR: 20,
URL_MISMATCH_ERR: 21,
QUOTA_EXCEEDED_ERR: 22,
TIMEOUT_ERR: 23,
INVALID_NODE_TYPE_ERR: 24,
DATA_CLONE_ERR: 25
```

It's not immediately clear whether `VALIDATION_ERROR` is, in fact, the correct error code to use.

Reply:
Chrome uses `INVALID_CHARACTER_ERR`

Reply:
I just noticed this...

```
try { atob('a'); } catch (err) { console.error(err.code); }
VM60:1 5
(anonymous) @ VM60:1
undefined
try { atob('a'); } catch (err) { console.error(err.name); }
VM108:1 InvalidCharacterError
```

Thanks @targos! Updating...

Reply:
For completeness sake: Safari too is using `INVALID_CHARACTER_ERR`.

Reply:
https://html.spec.whatwg.org/multipage/webappapis.html#dom-atob step 2